res/notes/modular-auth.md

```diff
@@ -0,0 +1,131 @@
+## Modular Auth notes
+
+- Authentication service
+    - Each service will have it's own repository to hold implementation
+    
+    - Things to consider
+        - ui
+            - Forms or any view component should be defined for each component.
+            - Idealy accounts/client will passed on the ui config method and it will then be used by the ui components 
+        - data store
+            - Store auth details for document based stores
+                - How indexes are handled?
+            - map to relational entries (for sql implmentation)
+        - added endpoints
+            - should expose the express/koa/hapi middleware if needed.
+    
+    - Register service plugins
+        - Register on client
+        - Register on server
+        - Suggested API
+        ```typescript
+        // client code
+        const accountsClient = new AccountsClient({
+            authServices: {
```

Comment:
Another approach is to have the client do a query to the server to determine which services have been configured.